pkg/controller/factory/factory.go

```diff
@@ -0,0 +1,67 @@
+package factory
+
+import (
+	"time"
+
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// Factory is generator that generate standard Kubernetes controllers.
+// Factory is really generic and should be only used for simple controllers that does not require special stuff..
+type Factory struct {
+	sync           SyncFunc
+	resyncInterval time.Duration
+	informers      []cache.SharedInformer
+	cachesToSync   []cache.InformerSynced
+}
+
+// New return new factory instance.
+func New() *Factory {
+	return &Factory{}
+}
+
+// Sync is used to set the controller synchronization function. This function is the core of the controller and is
+// usually hold the main controller logic.
+func (f *Factory) Sync(syncFn SyncFunc) *Factory {
+	f.sync = syncFn
+	return f
+}
+
+// Informers is used to register event handlers and get the caches synchronized functions.
+// Pass informers you want to use to react to changes on resources. If informer event is observed, then the Sync() function
+// is called.
+func (f *Factory) Informers(informers ...cache.SharedInformer) *Factory {
+	f.informers = informers
+	return f
+}
+
+// ResyncEvery will cause the Sync() function to be called periodically, regardless of informers.
+// This is useful when you want to refresh every N minutes or you fear that your informers can be stucked.
+// If this is not called, no periodical resync will happen.
+// Note: The controller context passed to Sync() function in this case does not contain the object metadata or object itself.
+//       This can be used to detect periodical resyncs, but normal Sync() have to be cautious about `nil` objects.
+func (f *Factory) ResyncEvery(interval time.Duration) *Factory {
+	f.resyncInterval = interval
+	return f
+}
+
+// Controller produce a runnable controller.
+func (f *Factory) Controller(name string, eventRecorder events.Recorder) Controller {
```

Comment:
I see, name is a string arg here. Then the sync context starts making more sense.

Reply:
I don't think we need controller name in Sync() anywhere, so I can just remove it :)